src/components/Truncate/Truncate.tsx

```diff
@@ -76,18 +76,30 @@ export class Truncate extends React.PureComponent<
       this.contentNode.style.display = 'initial'
       // 2. Calculate the differences
       const isContentTruncated =
-        // TODO: fix typescript complains
-        // @ts-ignore
-        this.contentNode.offsetWidth > this.node.offsetWidth
+        props.splitter && props.showTooltipOnTruncate
+          ? this.isSplitContentTruncated(this.contentNode, this.node)
+          : // TODO: fix typescript complains
+            // @ts-ignore
+            this.contentNode.offsetWidth > this.node.offsetWidth
       // 3. Resets the display
       // TODO: fix typescript complains
       // @ts-ignore
-      this.contentNode.style.display = null
 
       return isContentTruncated
     }
   }
 
+  isSplitContentTruncated = (contentNode: any, node: any): boolean => {
+    const [
+      firstSpan,
+      secondSpan,
+    ] = node.children[1].children[1].children[0].children[0].children[0].children
```

Comment:
We access this.node and this.contentNode so nicely using refs, is there a similar way to access these nested spans so that this isn't so ugly?

Reply:
I think I'm missing something here but why do we need to reach so deeply into the node?

Reply:
If the spans have classnames and node is an instance of HtmlElement can we use `node.querySelector` here instead?

Reply:
That's how deeply nested the two spans are that wrap the first and second chunks of the email. We need to get the scroll width of those two spans and compare it to the width of the parent span, and if the combined scroll width is larger than the parent, the text should be truncated. Good call with node.querySelector.